frontend/openchat-client/src/openchat.ts

```diff
@@ -4512,152 +4513,165 @@ export class OpenChat extends OpenChatAgentWorker {
+                };
+                userStore.add(this.rehydrateDataContent(user, "avatar"));
+            }
 
-                // If the latest message in a chat is sent by the current user, then we know they must have read up to
-                // that message, so we mark the chat as read up to that message if it isn't already. This happens when a
-                // user sends a message on one device then looks at OpenChat on another.
-                for (const chat of chats) {
-                    const latestMessage = chat.latestMessage?.event;
-                    if (
-                        latestMessage !== undefined &&
-                        latestMessage.sender === this._liveState.user.userId &&
-                        (chat.membership?.readByMeUpTo ?? -1) < latestMessage.messageIndex &&
-                        !unconfirmed.contains({ chatId: chat.id }, latestMessage.messageId)
-                    ) {
-                        messagesRead.markReadUpTo({ chatId: chat.id }, latestMessage.messageIndex);
-                    }
+            // If the latest message in a chat is sent by the current user, then we know they must have read up to
+            // that message, so we mark the chat as read up to that message if it isn't already. This happens when a
+            // user sends a message on one device then looks at OpenChat on another.
+            for (const chat of chats) {
+                const latestMessage = chat.latestMessage?.event;
+                if (
+                    latestMessage !== undefined &&
+                    latestMessage.sender === this._liveState.user.userId &&
+                    (chat.membership?.readByMeUpTo ?? -1) < latestMessage.messageIndex &&
+                    !unconfirmed.contains({ chatId: chat.id }, latestMessage.messageId)
+                ) {
+                    messagesRead.markReadUpTo({ chatId: chat.id }, latestMessage.messageIndex);
                 }
+            }
 
-                chatsInitialised.set(true);
+            chatsInitialised.set(true);
 
-                this.dispatchEvent(new ChatsUpdated());
-            }
-        } catch (err) {
-            this.config.logger.error("Error loading chats", err as Error);
-            throw err;
-        } finally {
-            chatsLoading.set(false);
+            this.dispatchEvent(new ChatsUpdated());
         }
     }
 
+    private async loadChats() {
+        const init = this._liveState.chatsInitialised;
+        chatsLoading.set(!init);
+
+        const updateRegistryTask = !init ? this.updateRegistry() : undefined;
+
+        return new Promise<void>((resolve) => {
+            this.sendStreamRequest({
+                kind: "getUpdates",
+                initialLoad: !init,
+            })
+                .subscribe(async (resp) => {
+                    await this.loadChatsResponse(updateRegistryTask, init, resp as UpdatesResult);
+                    chatsLoading.set(!this._liveState.chatsInitialised);
```

Comment:
Why not just `false`?

Reply:
Is it something to do with there potentially being more coming?

Reply:
just in case the `chatsInitialised` doesn't get set for some reason i.e. `chatsInitialised` is the source of truth.